### PR TITLE
Adding release cadence docs

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
Members of our community often ask about our release cadence, so I have documented it for easy future reference.

### CODE REVIEW

File: docs/sources/release-notes/cadence.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+# Release cadence
+
+## Stable Releases
+
+Loki releases (this includes [Promtail](/clients/promtail), [Loki Canary](/operations/loki-canary/), etc) use the following
+naming scheme: `MAJOR`.`MINOR`.`PATCH`
+
+- `MAJOR` (roughly once a year): these releases include large new features & possible backwards-compatibility breaks
+- `MINOR` (roughly once a quarter): these releases include new features which do not break backwards-compatibility
```
Comment (slim-bean) [inline docs/sources/release-notes/cadence.md:15]: Depending on how we defined our stability contract (what does it include) this isn't entirely true.

We do make breaking changes in minor releases from time to time, mostly around configs.

I think the note below sums it up best, we will over communicate and do our best to make upgrades as painless as possible, always. But we may make some changes considered breaking in minor releases.

Comment (JStickler) [inline docs/sources/release-notes/cadence.md:15]:
```suggestion
- `MINOR` (roughly once a quarter): these releases include new features which do not break backwards-compatibility.
```

File: docs/sources/release-notes/cadence.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- `PATCH` (roughly once or twice a month): these releases include bug & security fixes which do not break backwards-compatibility
```
Comment (JStickler) [inline docs/sources/release-notes/cadence.md:16]:
```suggestion
- `PATCH` (roughly once or twice a month): these releases include bug & security fixes which do not break backwards-compatibility.
```

File: docs/sources/release-notes/cadence.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+> **NOTE:** While our naming scheme resembles [Semantic Versioning](https://semver.org/), at this time we do not strictly follow its
+guidelines to the letter. Our goal is to provide regular releases that are as stable as possible, and we take backwards-compatibility
+seriously. As with any software, always read the [release notes](/release-notes) and the [upgrade guide](/upgrading) whenever
+choosing a new version of Loki to install.
+
+New releases are based of a [weekly release](#weekly-releases) which we have vetted for stability over a number of weeks.
+
+We strongly recommend keeping up-to-date with patch releases as they are released. We post updates of new releases in the `#loki` channel
+of our [slack community](/community/getting-in-touch).
```
Comment (JStickler) [inline docs/sources/release-notes/cadence.md:26]:
```suggestion
of our [Slack community](/community/getting-in-touch).
```

**Review by JStickler — COMMENTED** [review]
[docs team]  Couple of small punctuation edits.

File: docs/sources/release-notes/cadence.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+You can find all of our releases [on GitHub](https://github.com/grafana/loki/releases) and on [Docker Hub](https://hub.docker.com/r/grafana/loki).
+
+## Weekly Releases
+
+Every Monday morning, we create a new "weekly" release from the tip of the [`main` branch](https://github.com/grafana/loki).
+These releases use the following naming scheme:
+
+`weekly-k<week-number>` where `<week-number>` is the number of weeks since we began this process (2020-07-06).
```
[the remark this answers is not in the document]
Comment (dannykopping) [reply docs/sources/release-notes/cadence.md:35]: Can do; do you want it as a quoted section or just a plain indent?

[the remark this answers is not in the document]
Comment (JStickler) [reply docs/sources/release-notes/cadence.md:35]: I think just indented is fine.

File: docs/sources/release-notes/cadence.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+description: How our release process works
+weight: 1
+---
+
+# Release cadence
+
+## Stable Releases
+
+Loki releases (this includes [Promtail](/clients/promtail), [Loki Canary](/operations/loki-canary/), etc) use the following
+naming scheme: `MAJOR`.`MINOR`.`PATCH`
```
Comment (JStickler) [inline docs/sources/release-notes/cadence.md:12]:
```suggestion
naming scheme: `MAJOR`.`MINOR`.`PATCH`.
```

**Review by JStickler — APPROVED** [review]
[docs team]  LGTM

**Review by slim-bean — APPROVED** [review]
With the suggested edit, this looks good to me!

**Pull request merged**